connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -217,6 +217,9 @@ protected void finalOffsetCommit(boolean failed) {
         if (failed) {
             log.debug("Skipping final offset commit as task has failed");
             return;
+        } else if (isCancelled()) {
+            log.debug("Skipping final offset commit as task has been cancelled and its producer has already been closed");
```

Comment:
Is it necessarily true that the producer is already closed, given that the closure on cancellation is actually delegated to an executor?

Reply:
That's fair; technically, it may not be closed if we hit this part right after `cancelled` is flipped to `true` but not before the producer is actually closed. I'd personally err on the side of skipping the offset commit if the task is cancelled since it's scary to see offset commit failure messages and the odds of an offset commit being viable once we know the task has been cancelled are low, but I'm not too attached to this (especially on a flaky test PR). What are your thoughts?

FWIW, I've also updated the log message to remove the explanation of why we're skipping the commit; the detail about the producer being closed doesn't seem strictly necessary.